File: src/main/java/DiscourseLauncher.java
import com.bulenkov.darcula.DarculaLaf;
import com.manabreakstudios.discourse.ui.Theme;
import com.manabreakstudios.discourse.ui.app.nodes.ReplyChoiceNode;
import com.manabreakstudios.discourse.ui.core.nodeeditor.NodeEditor;
import com.manabreakstudios.discourse.ui.core.nodeeditor.NodeUI;

import javax.swing.*;
import java.awt.*;
import java.io.IOException;
import java.io.InputStream;

public class DiscourseLauncher {

    public static void main(String[] args) throws UnsupportedLookAndFeelException, IOException, FontFormatException {
        initLookAndFeel();

        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
        frame.setSize(800, 600);
        frame.setLocationRelativeTo(null);
        frame.setVisible(true);
        frame.setTitle("Discourse 0.1");

        frame.getContentPane().setLayout(new BorderLayout());

        NodeEditor nodeEditor = new NodeEditor();

        NodeUI nodeA = new NodeUI(new ReplyChoiceNode());
        nodeEditor.addNode(nodeA, 10, 10);

        NodeUI nodeB = new NodeUI(new ReplyChoiceNode());
        nodeEditor.addNode(nodeB, 300, 20);

        NodeUI nodeC = new NodeUI(new ReplyChoiceNode());
        nodeEditor.addNode(nodeC, 300, 200);

        nodeEditor.connect(nodeA.getSlot(1), nodeB.getSlot(0));
        nodeEditor.connect(nodeA.getSlot(2), nodeC.getSlot(0));

        frame.getContentPane().add(nodeEditor);
        frame.invalidate();
    }

    private static void initLookAndFeel() throws UnsupportedLookAndFeelException, FontFormatException, IOException {
        DarculaLaf.initInputMapDefaults(UIManager.getDefaults());
        UIManager.setLookAndFeel(new DarculaLaf());
        loadFont("Font.OpenSans-ExtraBold", "fonts/OpenSans-ExtraBold.ttf");
        loadFont("Font.OpenSans-Regular", "fonts/OpenSans-Regular.ttf");
        loadFont("Font.FontAwesome", "fonts/fontawesome-webfont.ttf");

        Theme.initialize(UIManager.getDefaults());
    }

    private static void loadFont(String resourceName, String filePath) throws FontFormatException, IOException {
        InputStream inputStream = DiscourseLauncher.class.getResourceAsStream(filePath);
        Font font = Font.createFont(Font.TRUETYPE_FONT, inputStream);
        UIManager.getDefaults().put(resourceName, font);
    }
}

File: app/src/com/manabreak/discourse/app/DiscourseLauncher.kt
import com.manabreak.discourse.nodes.ReplyChoiceNode
import com.manabreak.discourse.nodes.SequenceNode
import com.manabreak.node_editor.ui.NodeEditor
import com.manabreak.node_editor.ui.NodeUI
import javafx.application.Application
import javafx.scene.Scene
import javafx.scene.text.Font
import javafx.stage.Stage
import java.awt.FontFormatException
import java.io.IOException
import javax.swing.SwingUtilities
import javax.swing.UnsupportedLookAndFeelException

object DiscourseLauncher {

    @Throws(UnsupportedLookAndFeelException::class, IOException::class, FontFormatException::class)
    @JvmStatic fun main(args: Array<String>) {
        Application.launch(DiscourseApplication::class.java, *args)
    }

    class DiscourseApplication : Application() {

        override fun start(stage: Stage?) {

            loadFont("fontawesome-webfont")
            loadFont("OpenSans-ExtraBold")

            val nodeEditor = NodeEditor()

            val nodeA = NodeUI(ReplyChoiceNode())
            nodeEditor.addNode(nodeA, 10.0, 10.0)

            val nodeB = NodeUI(ReplyChoiceNode())
            nodeEditor.addNode(nodeB, 300.0, 20.0)

            val nodeC = NodeUI(ReplyChoiceNode())
            nodeEditor.addNode(nodeC, 300.0, 200.0)

            val nodeS = NodeUI(SequenceNode())
            nodeEditor.addNode(nodeS, 500.0, 200.0)

            SwingUtilities.invokeLater {
                nodeEditor.linkManager.link(nodeA.content.firstOutput, nodeB.content.input)
                nodeEditor.linkManager.link(nodeA.content.secondOutput, nodeC.content.input)
                nodeEditor.refresh()
            }

            stage!!.scene = Scene(nodeEditor)
            stage.title = "Discourse 1.0"
            stage.scene.stylesheets.add("css/style.css")
            stage.width = 800.0
            stage.height = 600.0
            stage.show()
        }

        private fun loadFont(fontName: String) {
            val url = DiscourseApplication::class.java.classLoader.getResource("fonts/$fontName.ttf").toExternalForm()
            Font.loadFont(url, 20.0)
        }
    }
}
